src/shared/components/results/DidYouMean.tsx

```diff
@@ -0,0 +1,171 @@
+  suggestions: Suggestion[];
+};
+
+const TIMEOUT = 3_000;
+
+type DidYouMeanProps = {
+  suggestions?: Suggestion[];
+};
+
+const DidYouMean = ({ suggestions }: DidYouMeanProps) => {
+  const currentNamespace = useNS();
+  const location = useLocation();
+  // Blocks a render until we have all network results, or we have timed out
+  const [hasTimedOut, setHasTimedOut] = useSafeState(false);
+  const otherNamespaceSuggestions = useRef<NamespaceSuggestions[]>([]);
+
+  const { query } = parseQueryString(location.search);
+
+  useEffect(() => {
+    if (!query || !currentNamespace) {
+      return;
+    }
+
+    const otherNamespaces = didYouMeanNamespaces.filter(
+      (ns) => ns !== currentNamespace
+    );
+
+    const promises = otherNamespaces.map((ns) =>
+      fetchData<{
+        results: APIModel[];
+      }>(
+        queryString.stringifyUrl({
+          url: apiUrls.search(ns),
+          query: { query, size: 0 },
+        })
+      ).then((response) => {
+        const hits = +(response?.headers?.['x-total-records'] || 0);
+        if (hits) {
+          otherNamespaceSuggestions.current.push({
+            namespace: ns,
+            suggestions: [{ query, hits }],
+          });
+        }
+      })
+    );
+
+    // If all of the queries have finished, trigger a render even before timeout
+    Promise.all(promises).then(() => {
+      setHasTimedOut(true);
+    });
+  }, [currentNamespace, query, setHasTimedOut]);
+
+  // Trigger the render after a set timeout
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setHasTimedOut(true);
+    }, TIMEOUT);
+    return () => clearTimeout(timeout);
+  }, [setHasTimedOut]);
+
+  if (!hasTimedOut) {
+    return <Loader />;
+  }
+
+  const suggestionsSortedByHits = orderBy(suggestions, ['hits'], ['desc']);
+  const querySuggestions: NamespaceSuggestions[] = [
+    ...(suggestionsSortedByHits.length && currentNamespace
+      ? [
+          {
+            suggestions: suggestionsSortedByHits,
+            namespace: currentNamespace as SearchableNamespace,
+          },
+        ]
+      : []),
+    ...orderBy(
+      otherNamespaceSuggestions.current,
+      ({ suggestions }) => suggestions[0].hits,
+      ['desc']
+    ),
+  ];
+
+  return (
+    <Message level="info" className={styles['did-you-mean-message']}>
+      <h4>Sorry, no results were found!</h4>
+      {!!querySuggestions.length && (
+        <div className={styles.suggestions}>
+          Did you mean to search for:
+          <ul className={styles['suggestions-list']}>
+            {querySuggestions.map(({ namespace, suggestions }) => (
+              <QuerySuggestionListItem
+                suggestions={suggestions}
+                namespace={namespace}
+                key={namespace}
+              />
+            ))}
+          </ul>
+        </div>
+      )}
+      Can&apos;t find what you are looking for? Please{' '}
+      <Link to={LocationToPath[Location.ContactGeneric]}>contact us</Link>
```

Comment:
It would need to pass the referrer to the contact page. Do check how it's done in other places in the codebase.

If you want, have a think about how to avoid having to remember to add the referrer everytime we link to the contact page page.